### PR TITLE
Add Inline Edit on Forms to Inline Edit pattern

### PR DESCRIPTION
**Pull request by zhutaoredhat**
Pull the Inline Edit files (Overview, Design, Mockups) to PatternFly
Design. https://github.com/patternfly/patternfly-design/issues/389

### CODE REVIEW

**Comment by zhutaoredhat** [comment]
Design Document:
https://docs.google.com/document/d/1Z94cNbL_8TJJZLPPWExRFkovYliID5WmEL9_-2nrjZ8/edit?ts=59e657e4#

**Comment by mcarrano** [comment]
One other thing, @zhutaoredhat . I think that including jump links at the top of the design page would be helpful in allowing the user to easily go to the section of interest.

File: pattern-library/forms-and-controls/inline-edit/design/design.md [hunk 1]
```diff
@@ -29,9 +29,26 @@
...
 
 ![Image of standard list view showing Edit action.](img/Table_02.png)
+
+
+## In-line Edit in a Form
+![Image of standard list view showing Edit action.](img/Form_Edit_Design.png)
+
+1. **Default Status:** The Information is displayed as text text without edit icon.
+2. **Status with Edit Icon:** If the most information can not be edited in the form, show the edit icon to the right of the field that can be edited which will help user easily  identify editable fields.
+3. **Hover Status:** When the mouse hover on the editable information, the edit icon (if not already shown)  and field border will appear. Click the text or edit button to make the field editable. If the text is a link, only click the edit icon can be turn to edit status. <br/>
```
[the remark this answers is not in the document]
Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: Sure. Thanks

File: pattern-library/forms-and-controls/inline-edit/design/overview.md [hunk 2]
```diff
@@ -20,3 +20,8 @@ In-line Edit can be used in conjunction with a List View or a Table view to allo
 1. This pattern applies to List and Table content views.
```
[the remark this answers is not in the document]
Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/overview.md]: I think so too, but not sure how to update it in before. Thank you~

**Comment by mindreeper2420** [comment]
The code for this design will be implemented/submitted to PatternFly through https://github.com/fabric8-ui/fabric8-ux/issues/709

**Comment by zhutaoredhat** [comment]
@mcarrano I had pulled the new commit, please help me review it~ Great thanks.

File: pattern-library/forms-and-controls/inline-edit/design/overview.md [hunk 3]
```diff
@@ -20,3 +20,8 @@ In-line Edit can be used in conjunction with a List View or a Table view to allo
...
 2. If drill-down behavior is supported, this pattern should not be used.
 3. This pattern does not include bulk actions (editing multiple rows at once).
+
+## In-Line Edit for Form
```
Comment (beanh66) [inline pattern-library/forms-and-controls/inline-edit/design/overview.md]: This section should be moved above the "Restrictions" section.

File: pattern-library/forms-and-controls/inline-edit/design/design.md [hunk 4]
```diff
@@ -1,12 +1,19 @@
 # In-line Edit
 
+- [In-line Create and Edit](#in-line-create-and-edit)
```
[the remark this answers is not in the document]
Comment (mcarrano) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: Yes, I agree @beanh66 @zhutaoredhat can you make that change? Also looks like there's a problem with some of these links.

File: pattern-library/forms-and-controls/inline-edit/design/overview.md (hunk 3 above)
Comment (mcarrano) [reply pattern-library/forms-and-controls/inline-edit/design/overview.md]: Yes, that makes sense.

File: pattern-library/forms-and-controls/inline-edit/design/design.md [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+- [In-line Edit in an Expanded List View](#in-line edit-in-an-expanded-list-view)
+- [In-line Edit in a Table View](#in-line-edit-in-a table-view)
+- [In-line Edit in a Form](#in-line-edit-in-a-form)
+
+
 ## In-line Create and Edit
 ![Image of list with newly created row.](img/List_Edit_01.png)
 
 1. **Create:** This action adds a row at the top of the list, in edit mode. All fields in the new row are editable. Default values for all fields are recommended.
 2. **Save and Cancel:** These buttons are available on the right end of the row. If user selects Save (check) without entering values for all fields an error is displayed. If user selects Cancel the new row is deleted.
```
[the remark this answers is not in the document]
Comment (mcarrano) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: Good catch @beanh66 . Yes, let's make this consistent.

File: pattern-library/forms-and-controls/inline-edit/design/design.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: @beanh66 @mcarrano That's great feedback!~ +1

File: pattern-library/forms-and-controls/inline-edit/design/design.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: +1

File: pattern-library/forms-and-controls/inline-edit/design/design.md [hunk 6]
```diff
@@ -29,9 +36,26 @@
...
   * When the editable row is the last one in the table, these buttons are displayed in a tab above the row.
 
 ![Image of standard list view showing Edit action.](img/Table_02.png)
+
+
+## In-line Edit in a Form
+![Image of standard list view showing Edit action.](img/Form_Edit_Design.png)
+
+1. **Default Status:** The Information is displayed as text text without edit icon.
+2. **Status with Edit Icon:** If the most information can not be edited in the form, show the edit icon to the right of the field that can be edited which will help user easily  identify editable fields.
```
[the remark this answers is not in the document]
Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: cool~

File: pattern-library/forms-and-controls/inline-edit/design/overview.md (hunk 3 above)
Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/overview.md]: Sure. The Restrictions section is to all inline edit. it should be updated.

File: pattern-library/forms-and-controls/inline-edit/design/design.md [hunk 7]
```diff
@@ -1,19 +1,20 @@
...
+Inline Edit has a number of use cases, jump to a section below for applied examples:
 
-- [In-line Create and Edit](#in-line-create-and-edit)
-- [In-line Edit in a List View](#in-line-edit-in a-list-view)
-- [In-line Edit in an Expanded List View](#in-line edit-in-an-expanded-list-view)
-- [In-line Edit in a Table View](#in-line-edit-in-a table-view)
-- [In-line Edit in a Form](#in-line-edit-in-a-form)
+- [Create and Edit](#inline-create-and-edit)
+- [List Views](#inline-edit-in-a-list-view)
+- [Expanded List Views](#inline edit-in-an-expanded-list-view)
```
Comment (beanh66) [inline pattern-library/forms-and-controls/inline-edit/design/design.md]: @zhutaoredhat link is broken, I think missing a dash.

Comment (zhutaoredhat) [reply pattern-library/forms-and-controls/inline-edit/design/design.md]: @beanh66 Thanks Colleen, the dash had been updated. I had checked many times before, but still lost this one~ Thanks for your mention~

**Comment by mcarrano** [comment]
Looks great @zhutaoredhat  @beanh66 and @catrobson if you can do one last review and approve, I will go ahead and merge.

**Comment by mcarrano** [comment]
This has been sitting for a while so I am going to merge. If there is a need for more work, we can make further updates in another PR. Thanks for your efforts on this @zhutaoredhat .

**Pull request merged**